errbot/backends/irc.py

```diff
@@ -248,6 +277,9 @@ def topic(self, topic):
         :param topic:
             The topic to set.
         """
+        if not self.joined:
+            raise RoomNotJoinedError("Must join the room to set the topic")
+        self.cb_set_topic(topic)
```

Comment:
It feels weird to call the callback function here, when changing the topic. Doesn't the IRC library trigger a `on_topic` event once the `self.connection.topic(self.room, topic)` is received and processed by the server? This line can be removed if that's the case.